src/common/components/markdownViewer/markdownViewerComponent.tsx

```diff
@@ -1,33 +1,109 @@
+      routerLocation: this.props.location ? this.props.location.pathname : '',
+      options: this.props.options,
+      codeStyle: this.props.codeStyle,
+    };
+    return CreateMarkdownRender(setupParams);
   }
 
-  return childrenComponent;
-};
+  private mdr: Mdr = this.CreateMdrInstance();
+  private scrollableContainerRef: HTMLElement = null;
+  private allowNotifyScroll: boolean = true;
+
+  private setScrollableContainerRef = (input) => { this.scrollableContainerRef = input; };
+
+  private markdownToHTML = () => {
+    return {
+      __html: this.mdr.render(this.props.content || ''),
+    };
+  }
+
+  private doScrollToSourceLine = (targetSourceLine) => {
+    const renderedElements = ReactDOM.findDOMNode(this).getElementsByClassName(SOURCE_LINE_CLASSNAME);
+    const scrollOffset = getPixelOffsetForSourceLine(renderedElements, targetSourceLine);
+    const componentPosition = this.scrollableContainerRef.getBoundingClientRect().top;
+    this.allowNotifyScroll = false;
+    this.scrollableContainerRef.scrollTop += scrollOffset - ((componentPosition > 0) ? componentPosition : 0);
+  }
+
+  private notifySourceLine = throttle(() => {
+    const componentPosition = this.scrollableContainerRef.getBoundingClientRect().top;
+    const renderedElements = ReactDOM.findDOMNode(this).getElementsByClassName(SOURCE_LINE_CLASSNAME);
+    const lineNum = getSourceLineForPixelOffset(renderedElements, componentPosition > 0 ? componentPosition : 0);
+    this.props.onScrollSourceLine(lineNum);
+  }, 25);
 
-export const MarkDownViewerComponent: React.StatelessComponent<MarkDownViewerComponentProps> = ({ content }) => {
-  return (
-    <div>
-      {getMarkDownChildren(content)}
-    </div>
-  );
+  private handleScroll = (event) => {
+    if (this.allowNotifyScroll) {
+      if (this.props.onScrollSourceLine) {
+        this.notifySourceLine();
+      }
+    } else {
+      this.allowNotifyScroll = true;
+    }
+  }
+
+  public shouldComponentUpdate(nextProps, nextState) {
+    if (nextProps.scrollSourceLine !== undefined && nextProps.scrollSourceLine !== this.props.scrollSourceLine) {
+      this.doScrollToSourceLine(nextProps.scrollSourceLine);
+      return false;
+    }
+    return true;
+  }
+
+  public componentWillUpdate(nextProps, nextState) {
+    // New render instance upon render options update.
+    if (nextProps.options !== this.props.options) {
+      this.mdr = this.CreateMdrInstance();
+    }
+  }
+
+  public render() {
+    const {className = ''} = this.props;
+    // WARNING: This conversion from plain HTML to JSX with
+    // dangerouslySetInnerHTML could be unsafe (script injection, XSS)
+    // depending whether markdown engine blocks malicious code or not.
+    // Markdonw-it is supposed to be XSS safe, but if you plan to
+    // change engine, ensure safety first!
+    return(
+      <div className={className} ref={this.setScrollableContainerRef}
+        dangerouslySetInnerHTML={this.markdownToHTML()}
```

Comment:
This make me scary "dangerously..." can we make sanitize it first? (if there is some easy way to do it)

Reply:
this issue needs review from your side @fjcalzado

Reply:
Yes @brauliodiez , I don't like `dangerously` either. That is a reminder from [React guys](https://reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml) to bear in mind that injecting HTML code directly to the React DOM could be risky if it is not sanitized first. You have to know what you are doing.

`markdown-it` provides 2 strategies to [make a safe output](https://github.com/markdown-it/markdown-it/blob/master/docs/security.md). I opted for the number 1. HTML won't be parsed, only syntax by default and those extra features extended with plugins. 

Just to put an example, [marsky](https://github.com/cerebral/marksy/blob/9fd762d584097a38f5cf3f35a56d1ca0111f42a1/src/createRenderer.js#L74), is also using `dangerouslySet...` to inyect its rendered HTML into React native components.

Anyway, we can open a new improvement for this if you guys feel better, and add an external sanitizer or research another solution.